pkg/types/baremetal/validation/platform.go

```diff
@@ -109,6 +109,10 @@ func validateOSImageURI(uri string) error {
 	} else {
 		return fmt.Errorf("the URI provided: %s must begin with http/https", uri)
 	}
+	// The URL will contain a checksum after the '?' so we can just check for the suffix followed by a '?'.
+	if !strings.Contains(uri, "qcow2.gz?") && !strings.Contains(uri, "qcow2.xz?") {
```

Comment:
Rather than relying on there being a query in the URL, why not take advantage of the having already parsed the URL into its parts?
```suggestion
	if _, file := filepath.Split(parsedURL.Path); !strings.Suffix(file, ".qcows.gz") && !strings.Suffix(file, ".qcows.xz") {
```

Reply:
Yeah that is a little better.  Fixed.  Thanks.

Reply:
It's nice that you left out the filepath splitting. That was clearly not necessary, and a remnant from my trying to use `filepath.Ext`.